charts/influxdb/templates/backup-configmap.yaml

```diff
@@ -0,0 +1,38 @@
+{{- if .Values.backup.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "influxdb.fullname" . }}-backup
+  labels:
+    {{- include "influxdb.labels" . | nindent 4 }}
+data:
+  backup.sh: |-
+    #!/bin/bash
+
+    set -e
+
+    DATE="$(date +%Y%m%d_%H%M%S)"
+
+    for DATABASE in $(influx -username {{ .Values.setDefaultUser.user.username }} -password "${INFLUXDB_ADMIN_USER_PASSWORD}" -host {{ include "influxdb.fullname" . }}.{{ .Release.Namespace }}.svc -port {{ .Values.config.http.bind_address | default 8086 }} -execute 'SHOW DATABASES' | sed -e '1,3d'); do
+        echo "backuping ${DATABASE} db to {{ .Values.backup.directory }}/${DATABASE}"
+        mkdir -p {{ .Values.backup.directory }}/${DATABASE}
+        influxd backup -host {{ include "influxdb.fullname" . }}.{{ .Release.Namespace }}.svc:{{ .Values.config.rpc.bind_address | default 8088 }} -portable -db ${DATABASE} {{ .Values.backup.directory }}/"${DATABASE}/${DATE}"
+    done
+
+    echo "deleting old backups"
+    find {{ .Values.backup.directory }} -not -name ".snapshot" -not -name "lost+found" -mindepth 2 -maxdepth 2 -type d -mtime +{{ .Values.backup.retentionDays }} -exec rm -r {} \;
+  upload-google.sh: |-
+    #!/bin/sh
+
+    set -ex
+    {{ if .Values.backup.uploadProviders.google.secret }}
+    gcloud auth activate-service-account --key-file /var/secrets/google/{{ .Values.backup.uploadProviders.google.secretKey }}
+    {{- end }}
+    gsutil -m rsync -r -d {{ .Values.backup.directory }}/ {{ .Values.backup.uploadProviders.google.bucketName }}
+  upload-azure.sh: |-
+    #!/bin/sh
+
+    set -ex
+
+    az storage blob sync --source {{ .Values.backup.directory }} --container {{ .Values.backup.uploadProviders.azure.containerName }}
+{{ end }}
```

Comment:
Can you please put these back in line in the cronjob to more easily compare any changes in the scripts?

Reply:
as dicsussed earlier i'd like to keep it that way as it makes everything much more readable.

Reply:
But you are also changing the nature of the scripts, it is more burdensome for review

Reply:
Very well, but it would be easier to review it if iterating on the following separately:
- extract the scripts
- change the scripts

Reply:
Pasting scripts here for easier comparison....


```sh
backup.sh: |-
    #!/bin/bash
    set -e
    DATE="$(date +%Y%m%d_%H%M%S)"
    for DATABASE in $(influx -username {{ .Values.setDefaultUser.user.username }} -password "${INFLUXDB_ADMIN_USER_PASSWORD}" -host {{ include "influxdb.fullname" . }}.{{ .Release.Namespace }}.svc -port {{ .Values.config.http.bind_address | default 8086 }} -execute 'SHOW DATABASES' | sed -e '1,3d'); do
        echo "backuping ${DATABASE} db to {{ .Values.backup.directory }}/${DATABASE}"
        mkdir -p {{ .Values.backup.directory }}/${DATABASE}
        influxd backup -host {{ include "influxdb.fullname" . }}.{{ .Release.Namespace }}.svc:{{ .Values.config.rpc.bind_address | default 8088 }} -portable -db ${DATABASE} {{ .Values.backup.directory }}/"${DATABASE}/${DATE}"
    done
```
vs
```sh
              influxd backup -host {{ include "influxdb.fullname" . }}.{{ .Release.Namespace }}.svc:{{ .Values.config.rpc.bind_address | default 8088 }} -portable /backup/$(date +%Y%m%d%H%M%S)
```

```
    echo "deleting old backups"
    find {{ .Values.backup.directory }} -not -name ".snapshot" -not -name "lost+found" -mindepth 2 -maxdepth 2 -type d -mtime +{{ .Values.backup.retentionDays }} -exec rm -r {} \;
```

vs not applicable, since backup file is transitory in emtpyDir in all cases

  upload-google.sh: |-
```
    set -ex
    {{ if .Values.backup.uploadProviders.google.secret }}
    gcloud auth activate-service-account --key-file /var/secrets/google/{{ .Values.backup.uploadProviders.google.secretKey }}
    {{- end }}
    gsutil -m rsync -r -d {{ .Values.backup.directory }}/ {{ .Values.backup.uploadProviders.google.bucketName }}
```
vs
```sh
              if [ -n "$KEY_FILE" ]; then
                gcloud auth activate-service-account --key-file $KEY_FILE
              fi
              gsutil -m cp -r /backup/* "$DST_URL"
```

  upload-azure.sh: |-

```sh
    #!/bin/sh
    set -ex
    az storage blob sync --source {{ .Values.backup.directory }} --container {{ .Values.backup.uploadProviders.azure.containerName }}
```

vs

```sh
              az storage container create --name "$DST_CONTAINER"
              az storage blob upload-batch --destination "$DST_CONTAINER" --destination-path "$DST_PATH" --source "$SRC_URL"
```

Reply:
So you still want separate pr's? 

Imho its not that hard to review.
You have an entirely new file here.
The file, the bash stuff was taken from, is missing some lines now.

Still think its a waste of time but if you insist i'll do it...

Reply:
Well I pasted the contents here to help me, but yeah it would make review much easier imho

Reply:
Tomorrow....

Reply:
No rush... Thanks 🙏 😅